bin/truffleruby.sh

```diff
@@ -12,17 +12,39 @@
 
 set -e
 
-function libext {
-    uname_str=$(uname)
-    if [ "$uname_str" = 'Linux' ] || [ "$uname_str" = 'SunOS' ]; then
-        echo so
-    elif [ "$uname_str" = 'Darwin' ]; then
-        echo dylib
-    else
-        echo "unknown platform $uname_str" 1>&2
-        exit 1
-    fi
-}
+# Check platform
+case $(uname) in
+  Linux)
+    os=linux
+    libext=so
+    ;;
+  Darwin)
+    os=darwin
+    libext=dylib
+    ;;
+  SunOS|Solaris)
+    os=solaris
+    libext=so
+    ;;
+  *)
+    echo "unknown platform $(uname)" 1>&2
+    exit 1
+    ;;
+esac
+
+# Check architecture
+case $(uname -m) in
+  x86_64)
```

Comment:
Can `uname -m` return amd64 on an AMD processor? (Questions like this is the reason why the original PR opted out to reuse mx.)

Reply:
I have not seen it on Linux/Darwin, if it does if will fail clearly and show the value we do not handle yet.
And it's very easy to add one.